haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs

```diff
@@ -506,7 +507,29 @@ refsFromRenamed ctx declAlts (hsGroup, _, _, _) =
         TypeSig names _ _ ->
 #endif
             mapMaybe (\(L l n) -> give ctx (nameLocToRef n TypeDecl l)) names
+        (PatSynSig lnames _) ->
+          lkupSigNames [lnames]
+        (FixSig (FixitySig lnames _)) ->
+          lkupSigNames lnames
+        (InlineSig lname _) -> lkupSigNames [lname]
+        (SpecSig lname ltypes _) -> lkupSigNames [lname]
+        (SpecInstSig st lst) -> []
+        (MinimalSig st lbf) -> lkupSigNames (namesFromLBooleanFormula lbf)
         _ -> []
+      where
+        lkupSigNames names =
+          mapMaybe (\(L l n) -> give ctx (nameLocToRef n Ref l)) names
+
+namesFromBooleanFormula :: BooleanFormula a -> [a]
+namesFromBooleanFormula bf =
+  case bf of
+    Var a -> [a]
```

Comment:
Optional: Travelsing this AST seems pretty compact, but consider using *plate. For example:

    let allNames = universeBi aBooleanFormula

This might also help unifying changes across GHC versions, trading a bit of performance (which could be gotten back if we would use `Data.Data.Lens` and pull in `lens`, but benchmarking needed).